puffin/src/frame_data.rs

```diff
@@ -428,6 +447,32 @@ impl FrameData {
         write.write_u8(packed_streams.compression_kind as u8)?;
         write.write_all(&packed_streams.bytes)?;
 
+        let mut to_serialize_scopes = Vec::new();
+
+        let scopes = if write_all_scopes {
+            scope_details
+                .scopes_by_id(|all_scopes| all_scopes.keys().cloned().collect::<HashSet<ScopeId>>())
```

Comment:
The keys should be `Copy`

```suggestion
                .scopes_by_id(|all_scopes| all_scopes.keys().copied().collect::<HashSet<ScopeId>>())
```